AirCasting/SessionViews/SessionCellView.swift

```diff
@@ -47,21 +55,39 @@ struct SessionCellView: View {
 }
 
 private extension SessionCellView {
+    
     var graphButton: some View {
-        NavigationLink(destination: GraphView(session: session, thresholds: Array(thresholds))) {
-            Text("graph")
+        Group {
+            if let threshold = thresholdFor(selectedStream: selectedStream) {
+                NavigationLink(destination: GraphView(session: session,
+                                                      threshold: threshold,
+                                                      selectedStream: $selectedStream)) {
+                    Text("graph")
+                }
+            }
         }
     }
     
     var mapButton: some View {
-        NavigationLink(destination: AirMapView(thresholds: Array(thresholds), session: session)) {
-            Text("map")
+        Group {
+            if let threshold = thresholdFor(selectedStream: selectedStream) {
+                NavigationLink(destination: AirMapView(threshold: threshold,
+                                                       session: session,
+                                                       selectedStream: $selectedStream)) {
+                    Text("map")
+                }
+            }
         }
     }
     
     func pollutionChart(stream: MeasurementStreamEntity) -> some View {
-        ChartView(stream: stream, thresholds: thresholds[0])
-            .frame(height: 200)
+        Group {
+            if let threshold = thresholdFor(selectedStream: selectedStream) {
```

Comment:
seams like evaluating `thresholdFor(selectedStream` is done many times. Please evaluate it once.